runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -212,12 +373,29 @@ private static void setListenerParamsForSnapshotSync(LogReplicationListener list
         }
     }
 
+    private static void setRoutingQueueListenerParamsForSnapshotSync(LogReplicationRoutingQueueListener listener,
```

Comment:
this method and updateListenerFlagsForSnapshotSync() perform the same operations.  Please consider reusing the existing methods.

Reply:
They have different listener. Are you suggesting to use a generic listener and type cast to specific one in a common method?